### PR TITLE
Test improvements

### PR DESCRIPTION
**Pull request by vaibhav-yb**
The tests in this project now work with `TestContainers` and can be simply run using `mvn test`.

### CODE REVIEW

File: src/test/java/io/debezium/connector/yugabytedb/TestHelper.java [hunk 1]
```diff
@@ -109,11 +157,21 @@ public static YugabyteDBConnectorConfig.LogicalDecoder decoderPlugin() {
...
-        return new YugabyteDBConnection(defaultJdbcConfig());
+        // return new YugabyteDBConnection(defaultJdbcConfig());
+        return new YugabyteDBConnection(defaultJdbcConfig(CONTAINER_YSQL_HOST, CONTAINER_YSQL_PORT));
     }
 
+    public static YugabyteDBConnection createConnectionTo(String databaseName) {
+        return new YugabyteDBConnection(defaultJdbcConfig(CONTAINER_YSQL_HOST, CONTAINER_YSQL_PORT, databaseName));
+    }
+
+    // public static YugabyteDBConnection create(String host, String ysqlPort) {
```
Comment (vrajat) [inline src/test/java/io/debezium/connector/yugabytedb/TestHelper.java:171]: Can you remove this? We can always add it later if required. Seems like a simple fn. and available in git history

File: src/test/java/io/debezium/connector/yugabytedb/TestHelper.java [hunk 2]
```diff
@@ -236,11 +298,106 @@ protected static Set<String> schemaNames() throws SQLException {
...
+            {
+                add(new PortBinding(Ports.Binding.bindPort(7100), new ExposedPort(7100)));
+                add(new PortBinding(Ports.Binding.bindPort(9100), new ExposedPort(9100)));
+                add(new PortBinding(Ports.Binding.bindPort(5433), new ExposedPort(5433)));
+                add(new PortBinding(Ports.Binding.bindPort(9042), new ExposedPort(9042)));
+            }
+        }));
+        container.withCommand("bin/yugabyted start --listen=0.0.0.0 --master_flags=rpc_bind_addresses=0.0.0.0 --daemon=false");
+        return container;
+        /*
```
Comment (vrajat) [inline src/test/java/io/debezium/connector/yugabytedb/TestHelper.java:345]: Delete the code. Available in history if required.

File: src/test/java/io/debezium/connector/yugabytedb/YugabyteDBConnectorIT.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.testcontainers.containers.YugabyteYSQLContainer;
+
+import io.debezium.DebeziumException;
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.embedded.AbstractConnectorTest;
+
+public class YugabyteDBConnectorIT extends AbstractConnectorTest {
```
Comment (vrajat) [inline src/test/java/io/debezium/connector/yugabytedb/YugabyteDBConnectorIT.java:21]: The class and file name has IT. Maven will consider it as an integration test. Is this different from DBCompleteTests? AFAICT, both start containers.

File: src/test/java/io/debezium/connector/yugabytedb/YugabyteDBDatatypesTest.java [hunk 4]
```diff
@@ -185,11 +205,15 @@ public void testVerifyValue() throws Exception {
...
         Thread.sleep(1000);
-        Configuration.Builder configBuilder = getConfigBuilder("public.t1");
+
+        String dbStreamId = TestHelper.getNewDbStreamId("yugabyte", "t1");
+        Configuration.Builder configBuilder = TestHelper.getConfigBuilder("public.t1", dbStreamId);
         start(YugabyteDBConnector.class, configBuilder.build());
         assertConnectorIsRunning();
         final long recordsCount = 1;
 
+        Thread.sleep(3000);
```
Comment (vrajat) [inline src/test/java/io/debezium/connector/yugabytedb/YugabyteDBDatatypesTest.java:215]: Is it possible to use awaitility with a condition? https://github.com/awaitility/awaitility

File: src/test/java/io/debezium/connector/yugabytedb/YugabyteDBDatatypesTest.java [hunk 5]
```diff
@@ -204,11 +228,15 @@ public void testNonPublicSchema() throws Exception {
...
         Thread.sleep(1000);
-        Configuration.Builder configBuilder = getConfigBuilder("test_schema.table_in_schema");
+
+        String dbStreamId = TestHelper.getNewDbStreamId("yugabyte", "table_in_schema");
+        Configuration.Builder configBuilder = TestHelper.getConfigBuilder("test_schema.table_in_schema", dbStreamId);
         start(YugabyteDBConnector.class, configBuilder.build());
         assertConnectorIsRunning();
         final long recordsCount = 1;
 
+        Thread.sleep(3000);
```
Comment (vrajat) [inline src/test/java/io/debezium/connector/yugabytedb/YugabyteDBDatatypesTest.java:238]: Possible to awaitility?

**Review by vrajat — CHANGES_REQUESTED** [review]
Can you also paste the o/p of mvn test? Just the test results part. I am curious if the IT test runs as part of test.

File: src/test/java/io/debezium/connector/yugabytedb/YugabyteDBDatatypesTest.java [hunk 6]
```diff
@@ -137,24 +133,43 @@ public void before() {
...
+    // This test will just verify that the TestContainers are up and running
+    // and it will also verify that the unit tests are able to make API calls.
+    @Test
+    public void testTestContainers() throws Exception {
+        TestHelper.dropAllSchemas();
+        TestHelper.executeDDL("postgres_create_tables.ddl");
+        Thread.sleep(1000);
+
+        insertRecords(2);
+        Thread.sleep(3000);
```
Comment (suranjan) [inline src/test/java/io/debezium/connector/yugabytedb/YugabyteDBDatatypesTest.java:150]: Why sleep?

File: src/test/java/io/debezium/connector/yugabytedb/YugabyteDBDatatypesTest.java [hunk 7]
```diff
@@ -137,24 +133,43 @@ public void before() {
...
-        Configuration.Builder configBuilder = getConfigBuilder("public.t1");
+
+        String dbStreamId = TestHelper.getNewDbStreamId("yugabyte", "t1");
+        Configuration.Builder configBuilder = TestHelper.getConfigBuilder("public.t1", dbStreamId);
         start(YugabyteDBConnector.class, configBuilder.build());
         assertConnectorIsRunning();
         final long recordsCount = 1;
 
-        System.out.println("testRecordConsumption");
+        Thread.sleep(3000);
```
Comment (suranjan) [inline src/test/java/io/debezium/connector/yugabytedb/YugabyteDBDatatypesTest.java:169]: why sleep?

File: src/test/java/io/debezium/connector/yugabytedb/YugabyteDBDatatypesTest.java [hunk 8]
```diff
@@ -167,13 +182,18 @@ public void testSmallLoad() throws Exception {
...
         Thread.sleep(1000);
-        Configuration.Builder configBuilder = getConfigBuilder("public.t1");
+
+        String dbStreamId = TestHelper.getNewDbStreamId("yugabyte", "t1");
+        Configuration.Builder configBuilder = TestHelper.getConfigBuilder("public.t1", dbStreamId);
         start(YugabyteDBConnector.class, configBuilder.build());
         assertConnectorIsRunning();
         final long recordsCount = 75;
 
+        Thread.sleep(3000);
```
Comment (suranjan) [inline src/test/java/io/debezium/connector/yugabytedb/YugabyteDBDatatypesTest.java:192]: why sleep?

File: src/test/java/io/debezium/connector/yugabytedb/YugabyteDBDatatypesTest.java (hunk 4 above)
Comment (suranjan) [inline src/test/java/io/debezium/connector/yugabytedb/YugabyteDBDatatypesTest.java:215]: why sleep? You have used sleep liberally, this is bad practice. Also, I don't understand, can you outline why we need it here? Is the database not ready?

**Comment by vaibhav-yb** [comment]
Merging now, will make the recommended changes in a new PR.

**Pull request merged**